applications/MappingApplication/custom_mappers/nearest_neighbor_mapper_iga.h

```diff
@@ -0,0 +1,299 @@
+* Each node on the destination side (FEM side) gets assigned is's closest integration point (neighbor) on the other side of the interface (IGA side).
+* Once the nearest integration point is found, the shape function values of the integration point are used to interpolate the values from the IGA side to the FEM side.
+* For information abt the available echo_levels and the JSON default-parameters
+* look into the class description of the MapperCommunicator
+*/
+template<class TSparseSpace, class TDenseSpace, class TMapperBackend>
+class KRATOS_API(MAPPING_APPLICATION) NearestNeighborMapperIGA
+    : public InterpolativeMapperBase<TSparseSpace, TDenseSpace, TMapperBackend>
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name Pointer Definitions
+    /// Pointer definition of NearestNeighborMapperIGA
+    KRATOS_CLASS_POINTER_DEFINITION(NearestNeighborMapperIGA);
+
+    typedef InterpolativeMapperBase<TSparseSpace, TDenseSpace, TMapperBackend> BaseType;
+    typedef typename BaseType::MapperUniquePointerType MapperUniquePointerType;
+    typedef typename BaseType::MapperInterfaceInfoUniquePointerType MapperInterfaceInfoUniquePointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Default constructor, needed for registration
+    NearestNeighborMapperIGA(ModelPart& rModelPartOrigin,
+                          ModelPart& rModelPartDestination)
+                          : BaseType(rModelPartOrigin, rModelPartDestination) {}
+
+    NearestNeighborMapperIGA(ModelPart& rModelPartOrigin,
+                          ModelPart& rModelPartDestination,
+                          Parameters JsonParameters)
+                          : BaseType(rModelPartOrigin,
+                                     rModelPartDestination,
+                                     JsonParameters)
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR_IF(!JsonParameters.Has("is_origin_iga") || JsonParameters["is_origin_iga"].GetBool() != true)
+            << "NearestNeighborMapperIGA expects the origin model part to be IGA.\n"
+            << "Please set \"is_origin_iga\": true in the mapper settings." << std::endl;
+
+        auto check_has_nodes = [](const ModelPart& rModelPart){
+            if (rModelPart.GetCommunicator().GetDataCommunicator().IsDefinedOnThisRank()) {
+                KRATOS_ERROR_IF(rModelPart.GetCommunicator().GlobalNumberOfNodes() == 0) << "No nodes exist in ModelPart \"" << rModelPart.FullName() << "\"" << std::endl;
+            }
+        };
+
+        auto check_has_conditions = [](const ModelPart& rModelPart){
+            if (rModelPart.GetCommunicator().GetDataCommunicator().IsDefinedOnThisRank()) {
+                KRATOS_ERROR_IF(rModelPart.GetCommunicator().GlobalNumberOfConditions() == 0) << "No conditions exist in ModelPart \"" << rModelPart.FullName() << "\"" << std::endl;
+            }
+        };
+
+        check_has_conditions(rModelPartOrigin);
+        check_has_nodes(rModelPartDestination);
```

Comment:
?
Why not just call whatever is inside, rather than defining the lambdas, invoking, then discarding them?